### DISCUSSION TITLE
Is there any document with the definition of keycloak roles?

### DISCUSSION THREAD

**Discussion by flixman** [keycloak/keycloak#29406] [Q&A] [answered]
I am trying to find a definition of the keycloak roles. E.g.: I want to retrieve the list of users and I have seen there are two roles available:
* view-users (which implies query users): enables the client to retrieve the list of users with some additional data.
* query-users: when used alone, and trying to get the list of users, returns 200 but the list is empty.

This for the endpoint "https://{host}/admin/realms/{realm}/users". I can I know which one allows me to get what I need, and nothing else?

**Comment by darius-m** [comment, marked as answer]
The roles are described in the [Server Administration guide](https://www.keycloak.org/docs/latest/server_admin/index.html#realm-specific-roles), but it looks like the list may not be complete (e.g., `view-users` is missing). From a comment in the [ClientPermissions.java](https://github.com/keycloak/keycloak/blob/6864ee0ead7c31badcd2f1d02d3c84034f7f8cd1/services/src/main/java/org/keycloak/services/resources/admin/permissions/ClientPermissions.java#L241) file, it seems like `view-users` may be relevant (only) when using fine-grained permissions, but I'm not sure. I see @pedroigor has added that comment, perhaps he could point us in the right direction?
